test/TestStepVestingPromo.js

```diff
@@ -37,25 +37,74 @@ contract('StepVesting', function ([_, owner, beneficiary]) {
       this.stepVestingPercent,
       this.numberOfPartitions,
       true
-      );
+    );
 
     await this.token.mint(this.vesting.address, amount, { from: owner });
 
     let va = await this.vesting.vestedAmount(this.token.address);
 
-	va.should.bignumber.equal(new BigNumber(0));
+	  va.should.bignumber.equal(new BigNumber(0));
 
-	//increase 2 sec
+	  //increase 2 sec
     const now = this.start + this.cliffDuration + 2;
 
     await timeUtils.increaseTimeTo(now);
 
-	va = await this.vesting.vestedAmount(this.token.address);
+	  va = await this.vesting.vestedAmount(this.token.address);
 
-	va.should.bignumber.equal(amount);
+	  va.should.bignumber.equal(amount);
 
     await this.vesting.release(this.token.address).should.be.fulfilled;
 
   });
 
+  it('can be released 50% after cliff and 50% after one single period', async function () {
+
+    this.token = await MintableToken.new({ from: owner });
+
+    this.start = timeUtils.latestTime() + timeUtils.duration.minutes(1); // +1 minute so it starts after contract instantiation
+    this.cliffDuration = timeUtils.duration.days(30);;
+    this.cliffPercent = 50;
+    this.stepVestingDuration = timeUtils.duration.days(30);;
+    this.stepVestingPercent = 50;
+    this.numberOfPartitions = 1;
+
+    this.vesting = await StepVesting.new(
+      beneficiary,
+      this.start,
+      this.cliffDuration,
+      this.cliffPercent ,
+      this.stepVestingDuration,
+      this.stepVestingPercent,
+      this.numberOfPartitions,
+      true
+    );
+
+    await this.token.mint(this.vesting.address, amount, { from: owner });
+
+    let va = await this.vesting.vestedAmount(this.token.address);
+
+    va.should.bignumber.equal(new BigNumber(0));
+
+    let now = this.start + this.cliffDuration + 2;
+
+    await timeUtils.increaseTimeTo(now);
+
+    va = await this.vesting.vestedAmount(this.token.address);
+
+    va.should.bignumber.equal(amount.div(2));
+
+    await this.vesting.release(this.token.address).should.be.fulfilled;
+
+    now = this.start + this.cliffDuration + this.stepVestingDuration + 2;
+
+    await timeUtils.increaseTimeTo(now);
+
+    va = await this.vesting.vestedAmount(this.token.address);
+
+    va.should.bignumber.equal(amount);
+
+    await this.vesting.release(this.token.address).should.be.fulfilled;
```

Comment:
lets add this -

    let fullReleased = await this.token.balanceOf(beneficiary);
    fullReleased.should.bignumber.equal(amount);

Reply:
done